--- a/stage2/builtins.c
+++ b/stage2/builtins.c
@@ -1494,7 +1494,7 @@ static struct builtin builtin_halt =
   halt_func,
   BUILTIN_CMDLINE | BUILTIN_HELP_LIST,
   "halt [--no-apm]",
-  "Halt your system. If APM is avaiable on it, turn off the power using"
+  "Halt your system. If APM is available on it, turn off the power using"
   " the APM BIOS, unless you specify the option `--no-apm'."
 };
 
@@ -3039,7 +3039,7 @@ static struct builtin builtin_password =
   " FILE as a new config file and restarts the GRUB Stage 2. If you"
   " omit the argument FILE, then GRUB just unlocks privileged"
   " instructions.  You can also use it in the script section, in"
-  " which case it will ask for the password, before continueing."
+  " which case it will ask for the password, before continuing."
   " The option --md5 tells GRUB that PASSWD is encrypted with"
   " md5crypt."
 };
