--- grub-0.96.orig/stage2/stage2.c
+++ grub-0.96/stage2/stage2.c
@@ -865,6 +865,14 @@
   return code_start;
 }
 
+static inline unsigned char * stack_ptr(void)
+{
+  unsigned char * u;
+
+  asm("movl %%esp, %0" : "=r" (u));
+
+  return u;
+}
 
 /*
  * Leave that much space on the heap. Everything else goes to the graphics
@@ -993,7 +1001,7 @@
 
   buf = (unsigned char *) (((unsigned) heap + 0xf) & ~0xf);
 
-  buf_size = (unsigned char *) &buf - buf - MIN_HEAP_SIZE;
+  buf_size = stack_ptr() - buf - MIN_HEAP_SIZE;
   buf_size &= ~0xf;
 
   /* too small */
@@ -1035,6 +1043,7 @@
   /* besides the file, we need some working memory, too */
   if(i + MIN_GFX_FREE + 0x0f >= (int) buf_size) {
     printf("file \"%s\" too large, press a key to continue...\n", graphics_file);
+    printf("i is %d, MIN_GFX_FREE is %d, buf_size is %d\n",i,MIN_GFX_FREE,buf_size);
     getkey();
     return;
   }
