--- a/stage2/bios.c
+++ b/stage2/bios.c
@@ -147,6 +147,7 @@ get_cdinfo (int drive, struct geometry *
   grub_memset (&cdrp, 0, sizeof (cdrp));
   cdrp.size = sizeof (cdrp) - sizeof (cdrp.dummy);
   err = biosdisk_int13_extensions (0x4B01, drive, &cdrp);
+  err = 0; /* really ugly hack to circumvent faulty BIOS versions like Acer 292LMi */
   if (! err && cdrp.drive_no == drive)
     {
       if ((cdrp.media_type & 0x0F) == 0)
